t/fits.t

```diff
@@ -2,340 +2,480 @@ use strict;
-	my @elem = sort keys %$table2;
-	my @expected = sort( qw( ACOL BCOL CCOL DCOL ECOL hdr tbl ) );
-	is_deeply \@elem, \@expected, "hash contains expected keys";
-
-	# convert the string array so that each element has the same length
-	# (and calculate the maximum length to use in the check below)
-	#
-	my $dlen = 0;
-	foreach my $str ( @$d ) {
-	  my $len = length($str);
-	  $dlen = $len > $dlen ? $len : $dlen;
-	}
-	foreach my $str ( @$d ) {
-	  $str .= ' ' x ($dlen-length($str));
-	}
-
-	# note that, for now, ushort data is written out as a long (Int4)
-	# instead of being written out as an Int2 using TSCALE/TZERO
-	#
-	my $i = 1;
-	foreach my $colinfo ( ( ["ACOL","1B",$x],
-				["BCOL","1I",$y],
-				["CCOL","1J",$c->long],
-				["DCOL","${dlen}A",$d],
-				["ECOL","1E",$e],
-	##			["FCOL","1M",$f]
-			      ) ) {
-	  is( $$table2{hdr}{"TTYPE$i"}, $$colinfo[0], "column $i is $$colinfo[0]" ); #37,43,49,55,58
-	  is( $$table2{hdr}{"TFORM$i"}, $$colinfo[1], "  and is stored as $$colinfo[1]" ); #38,44,50,56,59
-	  my $col = $$table2{$$colinfo[0]};
-	  if ( UNIVERSAL::isa($col,"PDL") ) {
-	    is_pdl $col, $$colinfo[2], $$colinfo[0]; #39-42,45-48,51-54,60-63
-	  } else {
-	    # Need to somehow handle the arrays since the data read in from the
-	    # file all have 15-character length strings (or whatever the length is)
-	    #
-	    is_deeply $col, $$colinfo[2], "$$colinfo[0] values agree (as an array reference)";
-	  }
-	  $i++;
-	}
-}
 ########### Check if r/wfits bugs are fixed ################
 
-{
-    local $| = 1;
-    my $a1 =  [1,2];
-    my $a2 = [[1,2],[1,2]];
-    my $p;
-    my $q;
-    my @target_bitpix = (8,16,32,-32,-64);
-    my $bp_i = 0;
-    for my $cref ( \(&byte, &short, &long, &float, &double) ) {
-        for my $x ($a1,$a2) {
-            $p = &$cref($x);
-            unlink $file;
-            $p->wfits($file);
-            $q = PDL->rfits($file);
-	    my $flag = 1;
-            if ( ${$p->get_dataref} ne ${$q->get_dataref} ) {
-	        $flag = 0;
-	        diag "\tnelem=",$p->nelem,"datatype=",$p->get_datatype;
-	        diag "\tp:", unpack("c" x ($p->nelem*howbig($p->get_datatype)), ${$p->get_dataref});
-	        diag "\tq:", unpack("c" x ($q->nelem*howbig($q->get_datatype)), ${$q->get_dataref});
+subtest 'Internal FITS Header regression tests' => sub {
+
+    local $PDL::Astro_FITS_HEADER = 0;
+
+    subtest 'r/wfits bug #1' => sub {
+        local $| = 1;
+        my $a1 = [ 1, 2 ];
+        my $a2 = [ [ 1, 2 ], [ 1, 2 ] ];
+        my $p;
+        my $q;
+        my @target_bitpix = ( 8, 16, 32, -32, -64 );
+        my $bp_i          = 0;
+        for my $cref ( \( &byte, &short, &long, &float, &double ) ) {
+            for my $x ( $a1, $a2 ) {
+                $p = &$cref($x);
+                my ( $fh, $file ) = tfile;
+                $p->wfits($file);
+                $q = PDL->rfits($file);
+                my $flag = 1;
+                if ( ${ $p->get_dataref } ne ${ $q->get_dataref } ) {
```

Comment:
A simple `is` here would suffice. Please use that.

Reply:
ok